SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs

```diff
@@ -0,0 +1,191 @@
+        }
+
+        private bool IsSlowCheetahInstalled(Project project)
+        {
+            var componentModel = (IComponentModel)this.package.GetService(typeof(SComponentModel));
+            IVsPackageInstallerServices installerServices = componentModel.GetService<IVsPackageInstallerServices>();
+            return installerServices.IsPackageInstalled(project, PackageName);
+        }
+
+        private bool IsSlowCheetahUpdated(Project project)
+        {
+            // Checks for older SC versions that require more complex update procedure
+            var componentModel = (IComponentModel)this.package.GetService(typeof(SComponentModel));
+            IVsPackageInstallerServices installerServices = componentModel.GetService<IVsPackageInstallerServices>();
+            IVsPackageMetadata scPackage =
+                    installerServices.GetInstalledPackages().First(pkg => string.Equals(pkg.Id, PackageName, StringComparison.OrdinalIgnoreCase));
+            if (scPackage != null)
+            {
+                Version ver;
+                if (Version.TryParse(scPackage.VersionString, out ver))
+                {
+                    return ver > new Version(2, 5, 15);
+                }
+            }
+
+            return false;
+        }
+
+        private void BackgroundInstallSlowCheetah(Project project)
+        {
+            string projName = project.UniqueName;
+            Task installTask;
+            if (!this.installTasks.TryGetValue(projName, out installTask))
+            {
+                if (this.HasUserAcceptedWarningMessage())
+                {
+                    // Gets the general output pane to inform user of installation
+                    IVsOutputWindowPane outputWindow = (IVsOutputWindowPane)this.package.GetService(typeof(SVsGeneralOutputWindowPane));
+                    if (outputWindow != null)
+                    {
+                        outputWindow.OutputString(string.Format("Installing SlowCheetah NuGet package to {0} ...\n", project.Name));
+                    }
+
+                    // Installs the latest version of the SlowCheetah NuGet package
+                    var componentModel = (IComponentModel)this.package.GetService(typeof(SComponentModel));
+                    IVsPackageInstaller2 packageInstaller = componentModel.GetService<IVsPackageInstaller2>();
+                    installTask = Task.Run(() =>
+                    {
+                        InstallSlowCheetahPackage(packageInstaller, project);
+                    }).ContinueWith(t =>
+                    {
+                        Task outTask;
+                        this.installTasks.TryRemove(projName, out outTask);
+                        if (outputWindow != null)
+                        {
+                            if (t.IsCompleted)
+                            {
+                                outputWindow.OutputString(string.Format("Finished installing SlowCheetah NuGet package to {0}.\n", project.Name));
```

Comment:
Typically calling `IVs*` interfaces on a background thread is a bad thing to do. I _think_ you are fine here because this is a fire & forget task, but I will double check to be sure.